addons/web_editor/models/ir_ui_view.py

```diff
@@ -80,10 +85,66 @@ def save_oe_structure(self, el):
             'mode': 'extension',
         }
         vals.update(self._save_oe_structure_hook())
-        self.env['ir.ui.view'].create(vals)
+        oe_structure_view = self.env['ir.ui.view'].create(vals)
+        self._copy_custom_snippet_translations(oe_structure_view, 'arch_db')
 
         return True
 
+    @api.model
+    def _copy_custom_snippet_translations(self, record, html_field):
+        """ Given a ``record`` and its HTML ``field``, detect any
+        usage of a custom snippet and copy its translations.
+        """
+        lang_value = record[html_field]
+        if not lang_value:
+            return
+
+        tree = html.fromstring(lang_value)
+        for custom_snippet_el in tree.xpath('//*[hasclass("s_custom_snippet")]'):
+            custom_snippet_name = custom_snippet_el.get('data-name')
+            custom_snippet_view = self.search([('name', '=', custom_snippet_name)], limit=1)
+            if custom_snippet_view:
+                self._copy_html_field_translations(custom_snippet_view, 'arch_db', record, html_field)
+
+    @api.model
+    def _copy_html_field_translations(self, record_from, field_from, record_to, field_to):
```

Comment:
maybe `_copy_field_model_terms_translations`. This function even doesn't require the `field_from.translate == field_to.translate`, but needs them to be `callable(field.translate)` and `field.store`